lib/src/test/java/io/ably/lib/test/realtime/RealtimeConnectFailTest.java

```diff
@@ -546,4 +555,198 @@ public void onConnectionStateChanged(ConnectionStateChange state) {
+                        retryCount.incrementAndGet();
+                    }
+                }
+            });
+
+            /* Inject detached message as if from the server */
+            ProtocolMessage detachedMessage = new ProtocolMessage() {{
+                action = Action.detached;
+                channel = channelName;
+                error = new ErrorInfo("Test error", errorCode);
+            }};
+            ably.connection.connectionManager.onMessage(null, detachedMessage);
+
+            /* wait for the client reattempt attachment */
+            channelWaiter.waitFor(ChannelState.attaching);
+
+            /* Inject detached+error message as if from the server */
+            ProtocolMessage errorMessage = new ProtocolMessage() {{
+                action = Action.detached;
+                channel = channelName;
+                error = new ErrorInfo("Test error", errorCode);
+            }};
+            ably.connection.connectionManager.onMessage(null, errorMessage);
+
+            int waitAtMost = 5 * 10; //5 seconds * 10 times per second
+            int waitCount = 0;
+            while (retryCount.get() < 6 && waitCount < waitAtMost) {
+                try {
+                    Thread.sleep(100);
+                    waitCount++;
+                } catch (InterruptedException e) {
+                    fail(e.getMessage());
+                }
+            }
+            System.out.println("wait done in: " + (waitCount / 10) + " seconds");
+
+            mockTransport.allowSend();
+
+            assertTrue("Disconnect retry was not finished, count was: " + retryCount.get(), retryCount.get() >= 6);
+
+            System.out.println("------------------------------------------------------------");
+            //check for all received retry times in onChannelStateChanged callback
+            //ignore first one as it is immediately done and the second one as it is close to our calculation
+            for (int i = 2; i < retryValues.size(); i++) {
+                long retryTime = retryValues.get(i);
+                long higherRange = channelRetryTimeout + Math.min(i, 3) * 50L * (i + 1);
+                double lowerRange = 0.6 * channelRetryTimeout + Math.min(i, 3) * 50;
+                System.out.println("higher range: " + higherRange + " - lower range: " + lowerRange + " | checked value: " + retryTime);
+
+                assertTrue("retry time higher range for count " + i + " is not in valid: " + retryTime + " expected: " + higherRange,
+                    retryTime < higherRange);
+                assertTrue("retry time lower range for count " + i + " is not in valid: " + retryTime + " expected: " + lowerRange,
+                    retryTime > lowerRange);
+            }
+            System.out.println("------------------------------------------------------------");
+        } catch (AblyException e) {
+            e.printStackTrace();
+            fail("init0: Unexpected exception instantiating library");
```

Comment:
The `try` block includes a lot more than just instantiating the library so maybe this fail message should be more generic?

Reply:
I thing including exception in text is best way to go https://github.com/ably/ably-java/pull/852/commits/688f4b5fe52f4e07b3ed2cc78514a35b62d305d7